packages/itwinui-css/src/surface/surface.scss

```diff
@@ -18,9 +18,36 @@
   border-radius: var(--iui-surface-border-radius);
   box-shadow: var(--iui-surface-elevation);
   color: var(--iui-color-text);
+
+  &:where([data-iui-layout='true']) {
+    display: flex;
+    flex-direction: column;
+  }
+}
+
+$surface-padding: var(--iui-size-2xs);
+
+@mixin iui-surface-header {
+  display: flex;
+  flex-shrink: 0;
+  align-items: center;
+  overflow: hidden;
+  border-bottom: 1px solid var(--iui-surface-border-color);
+  padding: $surface-padding;
+  scrollbar-gutter: stable both-edges;
 }
 
```

Comment:
I think everything looks good so I just have question for learning purpose when playing around in this branch :sweat_smile:

I tried to remove the paddings for this gear button because it is not balanced with the left side by adding
``
@mixin iui-button {
  padding: 4px 0;
}
``
here.

But nothing changed. How could I have achieved this?

(also image if you want to see what I'm talking about)
![image](https://user-images.githubusercontent.com/22462084/220764020-814f6dce-53d9-471a-b46a-1d617d489bcb.png)

Reply:
> I just have question for learning purpose when playing around in this branch 😅

Awesome, that's the best way to learn!

> I tried to remove the paddings for this gear button

If you wanted to make the button thinner but only when placed within the surface header, you'd need to also unset `min-width` which is currently set to be the height of the button (this makes the button 1/1 ratio at it's smallest size) :

```scss
@mixin iui-surface-header {
  … 

  .iui-button {
    padding: 0;
    min-width: auto;
  }
}
```

However, I would suggest against that.  All of our buttons, even [the borderless button](https://itwin.github.io/iTwinUI/css/button.html#demo-borderless), have space around the text / icon to increase the [touch target size](https://www.nngroup.com/articles/touch-target-size/).  A 36x36 px button is easier to tap with a finger than a 16x16 px icon.  [Accessibility best practices](https://www.w3.org/WAI/WCAG21/quickref/#target-size) suggest larger than our default 36px, it's something that we need to look into in the future.

Now, an alternative solution to visually align the icon without reducing the button size could be to set the header's left & right padding separately:
```
padding-left: 12px;
padding-right: 4px;
```
But we're not always 100% sure that the user might place a button in the right hand side.  They might place a checkbox with a label, or a badge:
![Screenshot 2023-02-22 at 4 55 21 PM](https://user-images.githubusercontent.com/849817/220768572-f4d5599d-2145-4481-a70b-9387f85f197a.png) ![Screenshot 2023-02-22 at 4 55 47 PM](https://user-images.githubusercontent.com/849817/220768576-2f5dc4d6-1ff5-4edb-89c3-1f8e6cd44b0e.png)

To avoid things getting too complex, we simply keep the padding equal on the left & right.

Reply:
Awesome. I just tested your suggestions and they work for what I had in mind. But I also understand we should keep things generic for users to meddle with. Thanks for explaining 😄